File: SpringBoot-Redis-Simple-Starter/src/main/java/com/example/springbootredissimplestarter/controller/OrderController.java
package com.example.springbootredissimplestarter.controller;


import com.example.springbootredissimplestarter.domain.Order;
import com.example.springbootredissimplestarter.service.OrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/order")
public class OrderController {

    @Autowired
    OrderService orderService;

    @PostMapping()
    public Order createOrder(@RequestBody Order order) {
        return orderService.createOrder(order);
    }

    @GetMapping()
    public ResponseEntity<List<Order>> getOrder() {

        return ResponseEntity.ok(orderService.getAllOrders());
    }

    @GetMapping("/{id}")
    public Order getOrder(@PathVariable Integer id) {
        return orderService.getOrder(id);
    }

    @PutMapping("/{id}")
    public Order updateOrder(@PathVariable Integer id, @RequestBody Order order) {
        return orderService.updateOrder(order, id);
    }

    @DeleteMapping("/{id}")
    public String deleteOrder(@PathVariable Integer id) {
        orderService.deleteOrder(id);

        return "Order with id: " + id + " deleted.";
    }
}

File: SpringKotlin-Simple-Example/src/main/kotlin/com/example/springkotlinsimpleexample/controller/OrderController.kt
package com.example.springkotlinsimpleexample.controller

import com.example.springkotlinsimpleexample.domain.dto.CreateOrderModelDTO
import com.example.springkotlinsimpleexample.service.OrderService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/api/v1/orders")
class OrderController {

    @Autowired
    private lateinit var orderService: OrderService

    @GetMapping(produces = ["application/json"])
    fun getOrders(): ResponseEntity<Any> {
        return ResponseEntity.ok(orderService.getAllOrders())
    }

    @GetMapping(value = ["/{orderUserName}"], produces = ["application/json"])
    fun getOrderByUserName(orderUserName: String): ResponseEntity<Any> {
        return ResponseEntity.ok(orderService.getOrderByOrderUserName(orderUserName))
    }

    @PostMapping()
    fun createOrder(@RequestBody createOrderDTO: CreateOrderModelDTO): ResponseEntity<Any> {
        orderService.createOrder(createOrderDTO)
        return ResponseEntity.ok().body(true)
    }
}
